crates/remote_server/src/unix.rs

```diff
@@ -1071,6 +1082,128 @@ impl ProtocolHandler for IrohZedProtocolHandler {
     }
 }
 
+struct IrohZedNode {
+    secret: SecretKey,
+    ticket: ZedIrohTicket,
+}
+
+impl IrohZedNode {
+    async fn create(persist: Option<&PathBuf>) -> Self {
+        if let Some(node_path) = persist {
+            match Self::read(&node_path).await {
+                Ok(Some(result)) => return result,
+                Ok(None) => {}
+                Err(error) => {
+                    log::error!("Error reading persisted Iroh Zed node: [{}]", error);
+                }
+            }
+        }
+        let iroh_zed_node = Self::generate();
+        if let Some(node_path) = persist {
+            if let Err(error) = Self::write(&node_path, &iroh_zed_node).await {
+                log::error!(
+                    "Could not persist Iroh Zed node: {:?}: {}",
+                    &node_path,
+                    error
+                );
+            }
+        }
+        iroh_zed_node
+    }
+
+    async fn read(node_path: &PathBuf) -> Result<Option<Self>> {
+        let (key_path, ticket_path) = split_node_path(&node_path);
+        if !key_path.exists() {
+            log::debug!("Secret key not found: {:?}", &key_path);
+            return Ok(None);
+        }
+        if !ticket_path.exists() {
+            log::debug!("Ticket not found: {:?}", &ticket_path);
+            return Ok(None);
+        }
+        let key_base64 = tokio::fs::read_to_string(key_path.clone()).await?;
+        let key_base64 = key_base64.trim();
+        let key_bytes = BASE64_STANDARD.decode(&key_base64)?;
+        if key_bytes.len() < 32 {
+            return Err(anyhow!("Not enough bytes in secret key"));
+        }
+        let secret = SecretKey::try_from(&key_bytes[0..32])?;
+        let public_key = secret.public();
+
+        let ticket_string = tokio::fs::read_to_string(ticket_path.clone()).await?;
```

Comment:
There is no need to persist the ticket, is actually potentially problematic.  The ticket contains the node id (public key) which will be already identical guaranteed if the secret key is identical. The rest of the information is currently available IP addresses & home relay, which have to get refreshed anyway.